Set up sphinx-apidoc to generate API docs

underyx opened the pull request:
<!--
Thanks for your contribution!

Reviewing pull requests take really a lot of time and we're all volunteers.
Please make sure you go through the following check list and complete
them all before pinging someone for a review.
-->
### Check list


- [x] I have rebased properly.
      [Please see this for a tutorial.](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing)
- [x] I have gone through the
      [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html)
      and I've followed them:
- [x] I have followed the
       [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the
       [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [ ] I have included the issue URL with the  `Fixes` keyword if the
       commit fixes a *real bug* and the `Closes` keyword if it adds a
       feature/enhancement.
       [See more here.](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference)
- [ ] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html)
      and they all pass. If any CI (below) is not green, please fix them. If
      GitMate issues appear you will have to amend your commits, pushing new
      commits on top is not sufficient!

### Reviewers
<!--
Please list below the Github handles of people you think suceptible to review
this PR
-->
@sils @Adrianzatreanu

<!--
End note:

As you learn things over your Pull Request please help others on the chat and
on PRs to get their stuff right as well!
-->

Adrianzatreanu commented on docs/index.rst:
```diff
@@ -1,5 +1,14 @@
+=======================
+
+Hey there! This is the documentation for developers of the coala project
+itself. If you're trying to **use** coala, you should doc have a look at our
```
"you should doc have a look" makes no sense to me ;d phrased it badly?

underyx replied:
Doc shouldn't be there; accidentally left it after a rewrite.

Adrianzatreanu commented:
It also seems that circleCI is failing due to you creating some moduels which arent included in the toctree (index.rst)

underyx commented:
@Adrianzatreanu yup, apparently `sphinx-apidoc` automatically creates a table of contents file (basically the same as the `index.rst` I added). I've added the `--no-toc` flag to disable this behavior.

sils commented:
ack https://github.com/coala/coala/pull/2969/commits/b3f66025288ba4b27bdd665febabb8f56b593757

sils commented:
@rultor merge

rultor commented:
> @rultor merge

@sils OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/9907-258776557)

rultor commented:
> @rultor merge

@sils Done! FYI, the full log is [here](http://www.rultor.com/t/9907-258776557) (took me 1min)
